numpy/conftest.py

```diff
@@ -136,3 +144,42 @@ def weak_promotion(request):
 
     yield request.param
     numpy._set_promotion_state(state)
+
+
+if HAVE_SCPDT:
+
+    @contextmanager
+    def warnings_errors_and_rng(test=None):
+        """Filter out the wall of DeprecationWarnings.
+        """
+        msgs = ["The numpy.linalg.linalg",
+                "The numpy.fft.helper",
+                "dep_util",
+                "pkg_resources",
+                "numpy.core.umath",
+                "msvccompiler",
+                "Deprecated call",
+                "numpy.core",
+                "`np.compat`",
+                "Importing from numpy.matlib"
+        ]
+
+        msg = "|".join(msgs)
+
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=DeprecationWarning, message=msg)
```

Comment:
The full tests have a check that doesn't like this, although the error mesage is bad, I think it's supposed to say something like we don't want to let code in that might filter a real warning. Obviously you need to ignore warnings to implement this context manager. Maybe there's a way to dynamically add `pytest.mark.filterwarnings` marks to a test?

Reply:
Let me do the simplest thing and skip the top level conftest.py in that test.